addons/s3_secret_handler.go

```diff
@@ -31,80 +25,43 @@ const (
 	DefaultS3Region = "noobaa"
 )
 
-func getFilterCondition(ownerReferences []metav1.OwnerReference, name string, blueSecretMatchString string) bool {
-	for _, ownerReference := range ownerReferences {
-		if ownerReference.Kind != ObjectBucketClaimKind {
-			continue
-		}
-		return strings.Contains(name, blueSecretMatchString)
-	}
-	return false
-}
-
-func (s3SecretHandler) getBlueSecretFilter(obj interface{}) bool {
-	blueSecretMatchString := os.Getenv("S3_EXCHANGE_SOURCE_SECRET_STRING_MATCH")
-	if blueSecretMatchString == "" {
-		blueSecretMatchString = utils.BucketGenerateName
-	}
-
-	if s, ok := obj.(*corev1.Secret); ok {
-		return getFilterCondition(s.OwnerReferences, s.ObjectMeta.Name, blueSecretMatchString)
-	} else if c, ok := obj.(*corev1.ConfigMap); ok {
-		return getFilterCondition(c.OwnerReferences, c.ObjectMeta.Name, blueSecretMatchString)
-	}
-
-	return false
-}
-
-func (s3SecretHandler) getGreenSecretFilter(obj interface{}) bool {
-	return false
-}
-
-func (s s3SecretHandler) syncBlueSecret(name string, namespace string, c *blueSecretTokenExchangeAgentController) error {
+func (r *S3SecretReconciler) syncBlueSecretForS3(ctx context.Context, name string, namespace string) error {
 	// cofig map and secret name and bucket claim name is same in nooba
 	// fetch obc secret
-	secret, err := getSecret(c.spokeSecretLister, name, namespace)
+	var secret corev1.Secret
+	err := r.SpokeClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: name}, &secret)
 	if err != nil {
 		return fmt.Errorf("failed to get the secret %q in namespace %q in managed cluster. Error %v", name, namespace, err)
 	}
-	isMatch := s.getBlueSecretFilter(secret)
-	if !isMatch {
-		// ignore handler which secret filter is not matched
-		return nil
-	}
 
 	// fetch obc config map
-	configMap, err := getConfigMap(c.spokeConfigMapLister, name, namespace)
+	var configMap corev1.ConfigMap
+	err = r.SpokeClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: name}, &configMap)
```

Comment:
is the configmap name same as the bucket name and namespace always? We were originally fetching all configmaps and filtering them if they belonged to the obc. 

Probably could check list the configmaps check their owner ref  for bucket name or through a particular label if possible